api/types/saml_idp_service_provider.go

```diff
@@ -161,6 +185,26 @@ func (s *SAMLIdPServiceProviderV1) CheckAndSetDefaults() error {
 		s.Spec.EntityID = ed.EntityID
 	}
 
+	if len(s.GetAttributeMapping()) > 0 {
+		attrNames := make([]string, 0)
+		for _, v := range s.GetAttributeMapping() {
+			// check for duplicate attribute names
+			if slices.Contains(attrNames, v.Name) {
+				return trace.Wrap(ErrDuplicateAttributeName)
+			}
+			attrNames = append(attrNames, v.Name)
+
+			// validate input name format. If it is using one of the
+			// supported values "unspecified", "basic" or "uri", update
+			// the value with respective uri format.
+			uriNameFormat, err := validateAndGetURINameFormat(v.NameFormat)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+			v.NameFormat = uriNameFormat
```

Comment:
Can we define `CheckAndSetDefaults()` method on `SAMLAttributeMapping` that does this and call it here instead?

Reply:
https://github.com/gravitational/teleport/pull/35584/commits/50d6feb921a700f1e4c9ab54b77b6e3d035c067f